lock

```diff
@@ -1,70 +1,90 @@
 #!/bin/bash
 
-#Constants
-DISPLAY_RE="([0-9]+)x([0-9]+)\\+([0-9]+)\\+([0-9]+)" # Regex to find display dimensions
-IMAGE_RE="([0-9]+)x([0-9]+)"    # Regex to find image dimensions
-FOLDER=`dirname "$BASH_SOURCE"` # Current folder
-CACHE_FOLDER="$FOLDER"/img/cache/ # Cache folder
+# Constants
+DISPLAY_RE="([0-9]+)x([0-9]+)\\+([0-9]+)\\+([0-9]+)"  # Regex to find display dimensions
+FOLDER=`dirname "$BASH_SOURCE"`  # Current folder
 
-#Image paths
-BKG_IMG="$FOLDER/img/background.png" # Path of background image
-MD5_BKG_IMG=$(md5sum $BKG_IMG | cut -c 1-10)
-TEXT_IMG="$FOLDER/img/text.png" # Path of text image
-MD5_TEXT_IMG=$(md5sum $TEXT_IMG | cut -c 1-10)
-MD5_SCREEN_CONFIG=$(xrandr | md5sum - | cut -c 1-32) # Hash of xrandr output
-OUTPUT_IMG="$CACHE_FOLDER""$MD5_SCREEN_CONFIG"."$MD5_BKG_IMG"."$MD5_TEXT_IMG".png # Path of final image
-OUTPUT_IMG_WIDTH=0 # Decide size to cover all screens
-OUTPUT_IMG_HEIGHT=0 # Decide size to cover all screens
+# Cache (create folder if not there yet)
+CACHE_FOLDER="$FOLDER"/image_cache/
+if ! [ -e $CACHE_FOLDER ]; then
+	mkdir $CACHE_FOLDER
+fi
+
+# Passed arguments
+while getopts ":i:a:" opt; do
+	case $opt in
+		i) arg_image="$OPTARG"
+		;;
+		a) lock_args="$OPTARG"
+		;;
+		\?) echo "Invalid option -$OPTARG" >&2 && exit 1
+		;;
+	esac
+done
+
+# Background image path
+if [ "$arg_image" ]; then
+	BKG_IMG="$arg_image"  # Passed image
+else
+	BKG_IMG="$FOLDER/background.png"  # Default
```

Comment:
this is nice! so users can override the background image with a cli parameter. But this includes a new location of the default background image which I won't merge since I want to keep the my image and my `img/` image folder.